extensions/podman/src/extension.ts

```diff
@@ -95,6 +98,27 @@ async function updateMachines(provider: extensionApi.Provider): Promise<void> {
       status = 'starting';
     }
 
+    let cpuUsage = 0;
+    let diskUsage = 0;
+    let memoryUsage = 0;
+    if (running) {
+      try {
+        const output = await execPromise(getPodmanCli(), [
+          '-c',
+          machine.Name,
+          'info',
+          '--format',
+          '{ "cpuUsage": {{.Host.CPUUtilization.IdlePercent}}, "memoryFree": {{.Host.MemFree}}, "memoryAllocated": {{.Host.MemTotal}}, "diskUsed": {{.Store.GraphRootUsed}}, "diskAllocated": {{.Store.GraphRootAllocated}} }',
+        ]);
+        const info = JSON.parse(output);
+        cpuUsage = 100 - info.cpuUsage;
+        diskUsage = (info.diskUsed * 100) / info.diskAllocated;
+        memoryUsage = ((info.memoryAllocated - info.memoryFree) * 100) / info.memoryAllocated;
+      } catch (err: unknown) {
+        console.error(` Can't get machine ${machine.Name} resource usage error ${err}`);
+      }
+    }
+
```

Comment:
would it be better to use the REST API instead ? so instead of executing a CLI and parsing JSON we would directly get JSON formatted response ?

Reply:
Yeah the reason is that the context is different (machine vs extensionAPI but will look if I can find a mapping between machine name and engineId

Reply:
yes parsing JSON output of a CLI would always lead to more errors than reusing the current connection (so no new process and we have direct JSON output) 👍